pkgs/tools/misc/z/default.nix

```diff
@@ -0,0 +1,45 @@
+{ fetchFromGitHub, lib, stdenv }:
+  stdenv.mkDerivation
+    { pname = "z";
+      version = "1622481819";
+      # ^ there isn't a version, so the UNIX time of the commit is used
+
+      src =
+        fetchFromGitHub
+          { owner = "rupa";
+            repo = "z";
+            rev = "b82ac78a2d4457d2ca09973332638f123f065fd1";
+            sha256 = "0phk9lswwxsypchb11qsnxy1pv5xg1zkrqj8im6x2icma63hfcz2";
+          };
+
+      phases = [ "unpackPhase" "installPhase" ];
+
+      installPhase =
+        ''
+        mkdir -p $out/share/man/man1
+        cp z.1 $out/share/man/man1
+        cp z.sh $out
+        # ^ for use in the module
+        '';
```

Comment:
Please use installManPage and comments go before the line they are about. Then the ^ is also not required. Also We don't install anything to /. Either use /bin or /share/z/ or similar.

Reply:
What would be the appropriate subdirectory to use for such a purpose? Or would it be better if I just exposed its path in `passthru`?

Reply:
How about ``${z}/bin/z.sh``?

Reply:
Well it's not an executable, that's the thing. It's a file that is sourced in a shell init script. I'm only putting it in the derivation so the module can get a hold of it easily.

Reply:
Then it should probably be under share/z/z.sh.